src/steps/2.addModules/generateIndexVue.ts

```diff
@@ -0,0 +1,172 @@
+import { moduleConfigs, Modules } from "./moduleConfigs"
+
+const getModulesSnippet = (selectedModules: Modules[], key: "html" | "css" | "js") => {
+  const moduleIndexSnippets = selectedModules.map((module) => moduleConfigs[module].indexVue?.[key]).filter(html => typeof html !== "undefined")
+  return moduleIndexSnippets.length > 0 ? "\n    " + moduleIndexSnippets.join("\n    ") : ""
+}
+export const generateIndexVue = (selectedModules: Modules[]) => {
+  const html = getModulesSnippet(selectedModules, "html")
+  const css = getModulesSnippet(selectedModules, "css")
+  const js = getModulesSnippet(selectedModules, "js")
+
+  return `<template>
+  <div class="main-container">
+    <div class="heading">
+      <h1 class="heading__title">
+        Welcome to your new <span class="gradient__text">sidebase</span> app!
+      </h1>
+      <p class="heading__credits">
+        Read our documentation <a href="https://sidebase.io/sidebase/welcome" target="_blank">here</a>. Get started in no time with the following amazing modules:
+      </p>
+    </div>
+    <div class="cards">${html}
+    </div>
+  </div>
+</template>
+
+<script>
```

Comment:
let's only add this section IFF `js` is non empty. else we'll have a template that contains an empty script tag in the end